starport/services/chain/chain.go

```diff
@@ -182,16 +185,105 @@ func (c *Chain) appVersion() (v version, err error) {
 	if err != nil {
 		return version{}, err
 	}
-	iter, err := repo.Tags()
+
+	tags, err := repo.Tags()
+	if err != nil {
+		return version{}, err
+	}
+
+	head, err := repo.Head()
+	if err != nil {
+		return version{}, err
+	}
+
+	cIter, err := repo.Log(&git.LogOptions{Order: git.LogOrderCommitterTime})
 	if err != nil {
 		return version{}, err
 	}
-	ref, err := iter.Next()
+
+	var (
+		tag             string
+		tagHash         string
+		commitIndex     int
+		taggerTimestamp int64
+	)
+
+	idMap := make(map[string]int)
+
+	err = cIter.ForEach(func(c *object.Commit) error {
+
+		idMap[c.Hash.String()] = commitIndex
+		commitIndex++
+
+		return nil
+	})
+
+	if err != nil {
+		return version{}, err
+	}
+
+	err = tags.ForEach(func(t *plumbing.Reference) error {
+		obj, err := repo.TagObject(t.Hash())
+
+		if err == nil {
+
+			_, exists := idMap[obj.Target.String()]
+
+			if !exists {
+				return nil
+			}
+
+			if taggerTimestamp < obj.Tagger.When.Unix() {
+				taggerTimestamp = obj.Tagger.When.Unix()
+
+				tag = strings.TrimPrefix(obj.Name, "v")
```

Comment:
maybe we shouldn't rm it anyway.

Reply:
Yep. I erased it at first, but when I took the cosmos sdk git repo and tested it, I found that ForEach through tags(), occurs not sorted well sometimes. So, it should be sorted based on the actual time written by Tagger.

```
// from https://github.com/cosmos/cosmos-sdk.git

it describe --tag
cosmovisor/v0.1.0-77-g9d58a5ad3

// code 
err = tags.ForEach(func(t *plumbing.Reference) error {
	fmt.Println(t.Name())
}

// the result is below
refs/tags/0.8.1
refs/tags/cosmovisor/v0.1.0
...
refs/tags/v0.43.0-alpha1
refs/tags/v0.43.0-beta1
refs/tags/v0.43.0-rc0
refs/tags/v0.43.0-rc1
refs/tags/v0.43.0-rc2
refs/tags/v0.43.0-rc3
refs/tags/v0.44.0
refs/tags/v0.5.0
refs/tags/v0.5.1
refs/tags/v0.5.2
refs/tags/v0.6.0
refs/tags/v0.6.1
refs/tags/v0.6.2
refs/tags/v0.7.0
refs/tags/v0.7.1
refs/tags/v0.8.0
refs/tags/v0.9.0

```
the commit code result is below
vcosmovisor/v0.1.0-g9d58a5ad

But, if I remove the code you mentioned, the result is below.
```
v0.43.0-alpha1-g9d58a5ad
```
 
So, I think the sorting code will be remain better.  How about your opinion?

Reply:
makes sense 👍